scope_registry.go

```diff
@@ -228,40 +232,48 @@ func (r *scopeRegistry) removeWithRLock(subscopeBucket *scopeBucket, key string)
 
 // Records internal Metrics' cardinalities.
 func (r *scopeRegistry) reportInternalMetrics() {
-	if r.skipInternalMetrics {
+	if r.internalMetricsOption != SendInternalMetrics {
 		return
 	}
 
 	counters, gauges, histograms := atomic.Int64{}, atomic.Int64{}, atomic.Int64{}
 	rootCounters, rootGauges, rootHistograms := atomic.Int64{}, atomic.Int64{}, atomic.Int64{}
-	r.ForEachScope(func(ss *scope) {
-		counterSliceLen, gaugeSliceLen, histogramSliceLen := int64(len(ss.countersSlice)), int64(len(ss.gaugesSlice)), int64(len(ss.histogramsSlice))
-		if ss.root { // Root scope is referenced across all buckets.
-			rootCounters.Store(counterSliceLen)
-			rootGauges.Store(gaugeSliceLen)
-			rootHistograms.Store(histogramSliceLen)
-			return
-		}
-		counters.Add(counterSliceLen)
-		gauges.Add(gaugeSliceLen)
-		histograms.Add(histogramSliceLen)
-	})
+	r.ForEachScope(
+		func(ss *scope) {
+			counterSliceLen, gaugeSliceLen, histogramSliceLen := int64(len(ss.countersSlice)), int64(len(ss.gaugesSlice)), int64(len(ss.histogramsSlice))
+			if ss.root { // Root scope is referenced across all buckets.
+				rootCounters.Store(counterSliceLen)
+				rootGauges.Store(gaugeSliceLen)
+				rootHistograms.Store(histogramSliceLen)
+				return
+			}
+			counters.Add(counterSliceLen)
+			gauges.Add(gaugeSliceLen)
+			histograms.Add(histogramSliceLen)
+		},
+	)
 
 	counters.Add(rootCounters.Load())
 	gauges.Add(rootGauges.Load())
 	histograms.Add(rootHistograms.Load())
 	log.Printf("counters: %v, gauges: %v, histograms: %v\n", counters.Load(), gauges.Load(), histograms.Load())
 
 	if r.root.reporter != nil {
-		r.root.reporter.ReportCounter(counterCardinalityName, internalTags, counters.Load())
-		r.root.reporter.ReportCounter(gaugeCardinalityName, internalTags, gauges.Load())
-		r.root.reporter.ReportCounter(histogramCardinalityName, internalTags, histograms.Load())
+		r.root.reporter.ReportCounter(r.root.sanitizer.Name(counterCardinalityName), internalTags, counters.Load())
```

Comment:
this will execute sanitazation on each flush, right?

Reply:
yes, I'll refactor this code into the constructor so it's only run once